vernac/vernacstate.ml

```diff
@@ -92,29 +95,6 @@ module System = struct
 
 end
 
-module Synterp = struct
-
-  type t =
-    { parsing : Parser.t
-    ; system : System.Synterp.t
-    }
-
-  let freeze () =
-    { parsing = Parser.cur_state ();
-      system = System.Synterp.freeze ();
-    }
-
-  let make_shallow s =
-    { s with system = System.Synterp.Stm.make_shallow s.system }
-
-  let init () = freeze ()
-
-  let unfreeze st =
-    System.Synterp.unfreeze st.system;
-    Pcoq.unfreeze st.parsing
-
```

Comment:
The ordering here is very interesting and maybe points to a previous problem that required `unfreeze` to be called twice due to other actions in the synterp summary?